### PR TITLE
New badguy: HangingBall

### PR DESCRIPTION
**Pull request by Hume2**
It's a forest equivalent of stalactite.

### CODE REVIEW

**Comment by Savsish** [comment]
Great graphics

**Comment by tobbi** [comment]
What about having a different sound for this instead of the cracking sound used for stalactites?

**Comment by Hume2** [comment]
@tobbi Do I look as someone who can hear properly?

**Comment by Hume2** [comment]
Sound taken from: http://freesound.org/people/juskiddink/sounds/140867/

**Comment by Hume2** [comment]
OK, the sound was Tobbi's joke. So it's a placeholder for now.

**Comment by maxteufel** [comment]
How about <https://freesound.org/people/kvgarlic/sounds/184547/> as sound?

File: src/badguy/stalactite.cpp [hunk 1]
```diff
@@ -36,7 +37,7 @@ Stalactite::Stalactite(const ReaderMapping& lisp) :
 {
   countMe = false;
   set_colgroup_active(COLGROUP_TOUCHABLE);
-  SoundManager::current()->preload("sounds/cracking.wav");
+  SoundManager::current()->preload(get_cracking_sound_path());
   SoundManager::current()->preload("sounds/sizzle.ogg");
   SoundManager::current()->preload("sounds/icecrash.ogg");
```
Comment (maxteufel) [inline src/badguy/stalactite.cpp]: The two other sounds should also get an option so they can be changed for Hanging Ball.

File: src/badguy/stalactite.cpp [hunk 2]
```diff
@@ -165,4 +166,91 @@ Stalactite::deactivate()
     remove_me();
 }
 
+std::string
+Stalactite::get_cracking_sound_path() const
+{
+  return "sounds/cracking.wav";
+}
+
+HangingBall::HangingBall(const ReaderMapping& reader) :
```
Comment (maxteufel) [inline src/badguy/stalactite.cpp]: Please use a file per class.

File: src/badguy/stalactite.hpp [hunk 3]
```diff
@@ -50,12 +50,38 @@ class Stalactite : public BadGuy
...
 
+  virtual std::string get_cracking_sound_path() const;
+
 protected:
   Timer timer;
   StalactiteState state;
   Vector shake_delta;
 };
 
+class HangingBall : public Stalactite
```
Comment (maxteufel) [inline src/badguy/stalactite.hpp]: Please use a file per class.

File: src/badguy/stalactite.cpp (hunk 2 above)
Comment (Hume2) [inline src/badguy/stalactite.cpp]: This is already in some other cases. This is not the only one.

File: src/badguy/stalactite.cpp (hunk 1 above)
Comment (Hume2) [inline src/badguy/stalactite.cpp]: The icecrash seems reasonable, but the sizzle is generic for melting process.

File: src/badguy/stalactite.cpp (hunk 2 above)
Comment (maxteufel) [inline src/badguy/stalactite.cpp]: That does not mean we should continue to do so.

**Comment by weluvgoatz** [comment]
a good concept, but hanging ball is not the most creative thing we could use.

**Comment by weluvgoatz** [comment]
Obviously, we are not going to use this.  Hume2 has not been active for a very long time.  Can someone close this?

**Pull request closed without merging**